File: trunk/openrest4cocoa/openrest4cocoa/openrest4cocoa-api/OpenRestDate.m
//
//  OpenRestDate.m
//  openrest4cocoa
//
//  Created by Yoav Amit on 5/31/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "OpenRestDate.h"


@implementation OpenRestDate

@synthesize year;
@synthesize month;
@synthesize day;
@synthesize hour;
@synthesize minute;

-(id)initWithDictionary:(NSDictionary*)data
{
    if ((self = [super init]))
    {
        year = [[data objectForKey:@"year"] intValue];
        month = [[data objectForKey:@"month"] intValue];
        day = [[data objectForKey:@"day"] intValue];
        hour = [[data objectForKey:@"hour"] intValue];
        minute = [[data objectForKey:@"minute"] intValue];
    }
    return self;
}

-(NSDictionary*)proxyForJson
{
    NSMutableDictionary* ret = [NSMutableDictionary dictionaryWithCapacity:0];

    [ret setValue:[NSNumber numberWithInt:year] forKey:@"year"];
    [ret setValue:[NSNumber numberWithInt:month] forKey:@"month"];
    [ret setValue:[NSNumber numberWithInt:day] forKey:@"day"];
    [ret setValue:[NSNumber numberWithInt:hour] forKey:@"hour"];
    [ret setValue:[NSNumber numberWithInt:minute] forKey:@"minute"];
    
    return ret;
}

-(NSString*)description
{
    return [NSString stringWithFormat:@"%d/%d/%d %d:%d", day, month, year, hour, minute];
}

@end

File: openrest4js/openrest4js-api/OpenrestDate.js
function OpenrestDateFromObj(data)
{
    return new OpenrestDate(data.year, data.month, data.day, data.hour, data.minute);
}

function OpenrestDate(year, month, day, hour, minute) {
    this.year = year;
    this.month = month;
    this.day = day;
    this.hour = hour;
    this.minute = minute;
}

OpenrestDate.prototype.toOpenRestObj = function()
{
    var ret = {};

    if (typeof(this.year) != "undefined") ret['year'] = this.year;
    if (typeof(this.month) != "undefined") ret['month'] = this.month;
    if (typeof(this.day) != "undefined") ret['day'] = this.day;
    if (typeof(this.hour) != "undefined") ret['hour'] = this.hour;
    if (typeof(this.minute) != "undefined") ret['minute'] = this.minute;

    return ret;
}
